Astronomy/Astronomy/WHLPhotoController.m

```diff
@@ -0,0 +1,213 @@
+        [NSURLQueryItem queryItemWithName:@"api_key" value:apiKey]
+    ];
+
+    NSURL *requestURL = urlComponents.URL;
+
+    NSURLSessionTask *task = [NSURLSession.sharedSession dataTaskWithURL:requestURL completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        NSLog(@"Inside of fetchManifest method with url: %@", requestURL);
+
+        if (error) {
+            completionBlock(error);
+            return;
+        }
+
+        if (!data) {
+            completionBlock(errorWithMessage(@"Error receiving data from sol fetch request", 1));
+            return;
+        }
+
+        NSError *jsonError = nil;
+
+        NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&jsonError];
+
+        if (jsonError) {
+            completionBlock(jsonError);
+            return;
+        }
+
+        [self.photos removeAllObjects];
+
+        NSArray *photosArray = json[@"photos"];
+
+        for (NSDictionary *photoDict in photosArray) {
+            NSNumber *sol = photoDict[@"sol"];
+            if([sol isKindOfClass:[NSNull class]]) { sol = nil; }
+
+            NSDictionary *camera = photoDict[@"camera"];
+            NSString *cameraName = camera[@"name"];
+            if([cameraName isKindOfClass:[NSNull class]]) { cameraName = nil; }
+
+            NSString *cameraFullName = camera[@"full_name"];
+            if([cameraFullName isKindOfClass:[NSNull class]]) { cameraFullName = nil; }
+
+            NSDate *photoDate = [self.dateFormatter dateFromString:photoDict[@"earth_date"]];
+            if([photoDate isKindOfClass:[NSNull class]]) { photoDate = nil; }
+
+            NSURL *imgSrc = [NSURL URLWithString:photoDict[@"img_src"]];
+            if([imgSrc isKindOfClass:[NSNull class]]) { imgSrc = nil; }
+            if (imgSrc) {
+                NSURLComponents *components = [NSURLComponents componentsWithURL:imgSrc resolvingAgainstBaseURL:YES];
+                components.scheme = @"https";
+                imgSrc = components.URL;
+            }
+
+            NSDictionary *rover = photoDict[@"rover"];
+            NSNumber *roverID = rover[@"id"];
+            if([roverID isKindOfClass:[NSNull class]]) { roverID = nil; }
+
+            NSString *roverName = rover[@"name"];
+            if([roverName isKindOfClass:[NSNull class]]) { roverName = nil; }
+
+            if (sol && cameraName && cameraFullName && photoDate && imgSrc && roverID && roverName) {
+                Photo *newPhoto = [[Photo alloc] initWithSol:sol.intValue cameraName:cameraName cameraFullName:cameraFullName photoDate:photoDate imgSrc:imgSrc roverID:roverID.intValue roverName:roverName];
+                [self.photos addObject:newPhoto];
+            }
+#warning program won't alert when the nil checker and will move on to the next object in the array
+            // TODO:- Add else statement to handle an attribute being nil
+        }
```

Comment:
You may already be thinking about this, but it would probably be best to factor this out into a convenience initializer on the Photo class.